### PR TITLE
Llama: always convert the causal mask in the SDPA code path

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Removes the `if` condition to apply the used-defined `attention_mask` on `causal_mask`: it is not required for correctness, and it prevents correct left-padding behavior in compile mode (related PR: https://github.com/huggingface/transformers/pull/29374).

I could not observe any performance degradation with eager dynamic cache nor with compiled static cache 🙌

### CODE REVIEW

**Review by fxmarty — APPROVED** [review]
LGTM

**Review by amyeroberts — APPROVED** [review]
Thanks!
